Libraries/src/net/relinc/libraries/application/StrainGaugeOnBar.java

```diff
@@ -28,12 +41,45 @@ public String getNameForFile(){
 	
 	@Override
 	public String stringForFile(){
-		String contents = super.stringForFile();
-		contents += distanceToSampleDescrip + ":" + Double.toString(distanceToSample) + SPSettings.lineSeperator;
-		contents += specificNameDescrip + ":" + specificName + SPSettings.lineSeperator;
-		return contents;
+
+		return getJSONObject().toString();
+
 	}
-	
+
+	@Override
+	public JSONObject getJSONObject() {
+		JSONObject jsonObject=super.getJSONObject();
+		jsonObject.put(distanceToSampleDescrip,distanceToSample);
+		jsonObject.put(specificNameDescrip,specificName);
+		return jsonObject;
+	}
+
+	public void setParametersFromJSONStrainGaugeOnBar(String file) {
+		//super.setParametersFromJSONString(file);
+
+		JSONParser jsonParser = new JSONParser();
+		JSONObject jsonObject = null;
+		try {
+			jsonObject = (JSONObject) jsonParser.parse(file);
+		} catch (org.json.simple.parser.ParseException e) {
+			//TODO throw exception
+		}
+		setVariableJSONStrainGaugeOnBar(jsonObject);
+	}
+
+	public void setVariableJSONStrainGaugeOnBar(JSONObject jsonObject) {
+		Object temp_obj=jsonObject.get(distanceToSampleDescrip);
+
+		if(temp_obj  != null ) {
```

Comment:
Do we ever expect this case to happen? Just curious why the pattern is different than the one used in `Sample`